pkg/kubelet/cm/container_manager_stub.go

```diff
@@ -41,14 +41,16 @@ import (
 )
 
 type containerManagerStub struct {
+	ctx                                 context.Context
 	shouldResetExtendedResourceCapacity bool
 	extendedPluginResources             v1.ResourceList
 }
 
 var _ ContainerManager = &containerManagerStub{}
 
-func (cm *containerManagerStub) Start(_ context.Context, _ *v1.Node, _ ActivePodsFunc, _ GetNodeFunc, _ config.SourcesReady, _ status.PodStatusProvider, _ internalapi.RuntimeService, _ bool) error {
+func (cm *containerManagerStub) Start(ctx context.Context, _ *v1.Node, _ ActivePodsFunc, _ GetNodeFunc, _ config.SourcesReady, _ status.PodStatusProvider, _ internalapi.RuntimeService, _ bool) error {
 	klog.V(2).InfoS("Starting stub container manager")
+	cm.ctx = ctx
```

Comment:
[Context documentation](https://pkg.go.dev/context) says that storing context should be avoided:
```
Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx.
```

[The Go blog](https://go.dev/blog/context-and-structs) elaborates on this quite nicely, PTAL.

Reply:
Nice read, thanks. There's an argument to be made ineed about the tradeoff for the stub, preserving the backward compatibility vs making the currently trivial (and beneficially trivial) stub code simple vs adding more complexity. So I pivoted a bit more in the latest upload.

Reply:
Basically, the friction point is that adding the context to the `InternalContainerLifecycle` call is not optimal: this will be needed just and only in the `stub code`, the real impls don't need it.
So it boils down to what to store in the stub object and what not.